### PR TITLE
Revolve improvements

### PR DESCRIPTION
**Pull request by zenfiric**
This batch of improvements contain:
- [x] Enable configuration options to users
- [x] Create brain selector

### CODE REVIEW

File: revolve.py [hunk 1]
```diff
@@ -36,10 +36,10 @@ def write_stderr(self, data):
 
 if __name__ == "__main__":
     settings = parser.parse_args()
+    manager_settings = sys.argv[1:]
     supervisor = OnlineEvolutionSupervisor(
-        manager_cmd='python3',
-        # manager_args=['-u', os.path.join(here, "experiments/examples/manager.py")],
-        manager_args=['-u', settings.manager],
```
Comment (portaloffreedom) [inline revolve.py:42]: `-u` option is important or you don't have output from the manager

File: cpprevolve/revolve/gazebo/Types.h [hunk 2]
```diff
@@ -22,9 +22,6 @@
 
 #include <vector>
```
Comment (portaloffreedom) [inline cpprevolve/revolve/gazebo/Types.h:25]: `#include <memory>`

File: cpprevolve/revolve/gazebo/brains/Brain.h [hunk 3]
```diff
@@ -12,9 +12,9 @@
 * See the License for the specific language governing permissions and
 * limitations under the License.
 *
-* Description: Specifies a utility `Brain` base class. If your brain doesn't
```
Comment (portaloffreedom) [inline cpprevolve/revolve/gazebo/brains/Brain.h]: fix documentation

File: cpprevolve/revolve/gazebo/brains/NeuralNetwork.cpp [hunk 4]
```diff
@@ -105,15 +105,15 @@ NeuralNetwork::NeuralNetwork(
     {
       std::cerr << "Missing required neuron attributes (id or layer). '"
                 << std::endl;
-      throw std::runtime_error("Robot brain error");
```
Comment (portaloffreedom) [inline cpprevolve/revolve/gazebo/brains/NeuralNetwork.cpp]: brainsssssszz

File: cpprevolve/revolve/gazebo/brains/NeuralNetwork.cpp [hunk 5]
```diff
@@ -253,22 +253,22 @@ NeuralNetwork::NeuralNetwork(
...
-    positionMap_[neuronId] = outPos;
+    positionMap_[neuronId] = outputsIndex;
 
     // Offset with output neurons within params / types
-    auto pos = nOutputs_ + outPos;
+    auto pos = nOutputs_ + outputsIndex;
     neuronHelper(&params_[pos * MAX_NEURON_PARAMS],
                  &types_[pos],
                  neuronMap[neuronId]);
-    outPos++;
```
Comment (portaloffreedom) [inline cpprevolve/revolve/gazebo/brains/NeuralNetwork.cpp:271]: `++outputIndex`

File: cpprevolve/revolve/gazebo/plugin/RobotController.cpp [hunk 6]
```diff
@@ -191,7 +190,19 @@ void RobotController::LoadBrain(const sdf::ElementPtr _sdf)
...
+    brain_.reset(new NeuralNetwork(this->model_, brain, motors_, sensors_));
+  }
+  else if ("rlpower" == learner)
+  {
+    brain_.reset(new RLPower(this->model_, brain, motors_, sensors_));
+  }
+  else if ("diff_cpg" == learner)
+  {
+    brain_.reset(new DifferentialCPG(this->model_, brain, motors_, sensors_));
+  }
```
Comment (portaloffreedom) [inline cpprevolve/revolve/gazebo/plugin/RobotController.cpp:207]: else error

File: tools/compile_python [hunk 7]
```diff
@@ -1,9 +1,9 @@
 #!/bin/sh
```
Comment (portaloffreedom) [inline tools/compile_python]: rename file compile_python.sh

**Review by portaloffreedom — CHANGES_REQUESTED** [review]

File: cpprevolve/revolve/gazebo/Types.h (hunk 2 above)
Comment (zenfiric) [reply cpprevolve/revolve/gazebo/Types.h:25]: Added in #292cd7a commit.

File: cpprevolve/revolve/gazebo/brains/Brain.h (hunk 3 above)
Comment (zenfiric) [reply cpprevolve/revolve/gazebo/brains/Brain.h]: Fixed in daebd2d

File: cpprevolve/revolve/gazebo/brains/NeuralNetwork.cpp (hunk 4 above)
Comment (zenfiric) [reply cpprevolve/revolve/gazebo/brains/NeuralNetwork.cpp]: Fixed in daebd2d

File: cpprevolve/revolve/gazebo/plugin/RobotController.cpp (hunk 6 above)
Comment (zenfiric) [reply cpprevolve/revolve/gazebo/plugin/RobotController.cpp:207]: Fixed in 05471c4

File: cpprevolve/revolve/gazebo/brains/NeuralNetwork.cpp (hunk 5 above)
Comment (zenfiric) [reply cpprevolve/revolve/gazebo/brains/NeuralNetwork.cpp:271]: Fix in 048524f

File: tools/compile_python (hunk 7 above)
Comment (zenfiric) [reply tools/compile_python]: Fixed in 53f9fef

**Review by portaloffreedom — APPROVED** [review]

**Pull request merged**